### ISSUE TITLE
Not Working Effectively with Navigation

### ISSUE THREAD

**Issue by s-doddapaneni** [NathanaelA/nativescript-platform-css#25]
None of the CSS is being constant in the entire application. 

It is going to default after a while or user click.

**Comment by NathanaelA** [comment]
Are you using Angular or Vue, or Standard NS?

**Comment by s-doddapaneni** [comment]
CSS is getting applied for the first time, if I navigate through my application or come back to login page by signing out. For the next login, the CSS is not getting applied to that. It is applying only for the first time when I build the application.

**Comment by NathanaelA** [comment]
Is this while just running the app; or when you are using LiveSync/HMR to update the app?

**Comment by s-doddapaneni** [comment]
Actually both. While running the app and also while using LiveSync/HMR

**Comment by NathanaelA** [comment]
LiveSync/HMR will typically overright the values frequently; that is not expected to work.   But if you started the app and are just navigating in the app (w/o ever using HMR/LiveSync) and it is failing then that is a problem.

**Comment by s-doddapaneni** [comment]
Does that problem is related to webpack ?

**Comment by NathanaelA** [comment]
Well then that is a problem with the plugin.   Any chance you can create a sample app that shows it failing.   Having a real app that fails is 99% of solving the issue.  :grinning:

**Comment by s-doddapaneni** [comment]
Yes. I can create a sample app and share that with you or may be I can
share it in playground as well.

**Comment by s-doddapaneni** [comment]
[SampleNS.zip](https://github.com/NathanaelA/nativescript-platform-css/files/2792679/SampleNS.zip)

PFA, CSS is not getting applied when you press back and go to the next screen again.

**Comment by s-doddapaneni** [comment]
Did you able to run that sample project. Did you face the same issue?

**Comment by NathanaelA** [comment]
Sorry I haven't had time yet; too many other paid projects that I've been working on.   It is in my list; but paid work always comes first...  :grinning:

**Comment by s-doddapaneni** [comment]
I found out the exact problem.
It is with the navigate method. Without using navigate, if I simply use this.routerExt.back(); for coming back. then it is working.

**Comment by danielnitu** [comment]
I can confirm this is an issue as well. Any element targeted with the library's classes will have the styling applied at first, but as soon as you navigate to another view and back the styling's gone.

For the moment I'm styling elements programatically from the component:
```typescript
import { Component, OnInit } from '@angular/core'

import {screen, isAndroid} from "tns-core-modules/platform/platform"
import { Page } from 'tns-core-modules/ui/page/page'

@Component({
        moduleId: module.id,
	selector: 'sign-in',
	templateUrl: './sign-in.component.html',
	styleUrls: ['./sign-in.component.css']
})
export class SignInComponent implements OnInit {

	constructor(
            private page: Page
        ) { }

	ngOnInit() {
            this.adaptLayout()
        }

        // Check the screen's width and hide the element on small screens
        private adaptLayout() {
            if (isAndroid) {
                if (screen.mainScreen.widthDIPs <= 360) {
                    this.page.addCss(".city { visibility: collapse; }")
                }
            }   else {
                if (screen.mainScreen.widthDIPs <= 320) {
                    this.page.addCss(".city { visibility: collapse; }")
                }
           }
        }
}
```
